arangod/Graph/Enumerators/WeightedTwoSidedEnumerator.cpp

```diff
@@ -0,0 +1,714 @@
+      _resultPath{_left.provider(), _right.provider()} {}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
+                           PathValidator>::~WeightedTwoSidedEnumerator() {}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+auto WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
+                                PathValidator>::destroyEngines() -> void {
+  // Note: Left & Right Provider use the same traversal engines.
+  //   => Destroying one of them is enough.
+  _left.provider().destroyEngines();
+}
+
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+void WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
+                                PathValidator>::clear() {
+  // Order is important here, please do not change.
+  // 1.) Remove current results & state
+  _candidates.clear();
+  _bestCandidateLength = -1.0;
+  _leftInitialFetch = false;
+  _rightInitialFetch = false;
+  _handledInitialFetch = false;
+
+  // 2.) Remove both Balls (order here is not important)
+  _left.clear();
+  _right.clear();
+
+  // 3.) Remove finished state
+  setAlgorithmUnfinished();
+}
+
+/**
+ * @brief Quick test if the finder can prove there is no more data available.
+ *        It can respond with false, even though there is no path left.
+ * @return true There will be no further path.
+ * @return false There is a chance that there is more data available.
+ */
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+bool WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
+                                PathValidator>::isDone() const {
+  return (_candidates.empty() && searchDone()) || isAlgorithmFinished();
+}
+
+/**
+ * @brief Reset to new source and target vertices.
+ * This API uses string references, this class will not take responsibility
+ * for the referenced data. It is caller's responsibility to retain the
+ * underlying data and make sure the strings stay valid until next
+ * call of reset.
+ *
+ * @param source The source vertex to start the paths
+ * @param target The target vertex to end the paths
+ */
+template<class QueueType, class PathStoreType, class ProviderType,
+         class PathValidator>
+void WeightedTwoSidedEnumerator<QueueType, PathStoreType, ProviderType,
+                                PathValidator>::reset(VertexRef source,
+                                                      VertexRef target,
+                                                      size_t depth) {
+  clear();
+
+  _left.reset(source, 0);
+  _right.reset(target, 0);
+  _resultPath.clear();
+
+  // Special depth == 0 case
+  if (_options.getMinDepth() == 0 && source == target) {
+    _left.testDepthZero(_right, _candidates);
+  }
+}
```

Comment:
Originally transferred from the "TwoSidedEnumerator". Info:

This must be checked before we merge back. 
The min and max values are being set here: 

ShortestPathNode:413
```
  arangodb::graph::TwoSidedEnumeratorOptions enumeratorOptions{
      0, std::numeric_limits<size_t>::max(),
      arangodb::graph::PathType::Type::ShortestPath};
```

Reply:
Currently the implementation does not do anything (blank body)